File: ontrack-extension-ldap/src/main/java/net/nemerosa/ontrack/extension/ldap/ExtendedLDAPUserDetails.java
package net.nemerosa.ontrack.extension.ldap;

import lombok.Data;
import lombok.experimental.Delegate;
import org.springframework.security.ldap.userdetails.LdapUserDetails;

import java.util.Collection;

@Data
public class ExtendedLDAPUserDetails implements LdapUserDetails {

    @Delegate
    private final LdapUserDetails support;
    private final String fullName;
    private final String email;
    private final Collection<String> groups;

}

File: ontrack-extension-ldap/src/main/java/net/nemerosa/ontrack/extension/ldap/ExtendedLDAPUserDetails.kt
package net.nemerosa.ontrack.extension.ldap

import org.springframework.security.ldap.userdetails.LdapUserDetails

open class ExtendedLDAPUserDetails(
        private val support: LdapUserDetails,
        val fullName: String,
        val email: String,
        val groups: Collection<String>
) : LdapUserDetails by support
